src/backends/postgresql/standard-use-type.cpp

```diff
@@ -159,7 +160,25 @@ void postgresql_standard_use_type_backend::pre_use(indicator const * ind)
                 snprintf(buf_, bufSize, "%lu", bbe->oid_);
             }
             break;
-
+        case x_xmltype:
+            {
+                xml_type * xml = static_cast<xml_type *>(data_);
+                std::string * s = &(xml->value);
+                
+                buf_ = new char[s->size() + 1];
+                std::strcpy(buf_, s->c_str());
+            }
+            break;
+        case x_longstring:
+            {
+                long_string * ls = static_cast<long_string *>(data_);
+                std::string * s = &(ls->value);
+                
+                buf_ = new char[s->size() + 1];
+                std::strcpy(buf_, s->c_str());
```

Comment:
This is a non-issue for several reasons.
1. Arbitrary binary data, with or without embedded zeros, can be stored as BLOBs. There is no reason to misuse strings for the same use case.
2. There is nothing special in long strings with regard to embedded zeros. The same question can be discussed in the context of short strings, too - and the last decade is an evidence that such a feature is not missing.
3. PostgreSQL client API, which is used below this layer, is string based and I'm not even sure if it would work with embedded zeros. Please investigate.